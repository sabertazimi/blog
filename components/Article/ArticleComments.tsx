'use client'

import { siteConfig } from '@/config'
import { useDisqus } from '@/hooks'
import cx from 'classnames'
import styles from './ArticleComments.module.css'

interface Props {
  url?: string
}

function ArticleComments({
  url = siteConfig.disqusUrl,
}: Props): JSX.Element {
  useDisqus(url)

  return (
    <div
      id="disqus_thread"
      className={cx(styles.disqus, 'after:bg-white dark:after:bg-black')}
    />
  )
}

export default ArticleComments
