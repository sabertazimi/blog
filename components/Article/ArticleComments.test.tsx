import { render } from '@utils'
import { axe } from 'jest-axe'
import ArticleComments from './ArticleComments'

describe('articleComments', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(<ArticleComments />)

    expect(container).toMatchSnapshot()
  })

  it('should render accessibility guidelines (AXE)', async () => {
    const { container } = render(<ArticleComments />)

    const a11y = await axe(container)

    expect(a11y).toHaveNoViolations()
  })
})
