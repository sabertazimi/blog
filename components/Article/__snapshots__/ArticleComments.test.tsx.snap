// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articleComments should render correctly (snapshot) 1`] = `
<div>
  <div
    class="disqus after:bg-white dark:after:bg-black"
    id="disqus_thread"
  />
</div>
`;
